tests/functional/base_test.go

```diff
@@ -0,0 +1,336 @@
+	raw := map[string]interface{}{
+		"apiVersion": "manila.openstack.org/v1beta1",
+		"kind":       "ManilaShare",
+		"metadata": map[string]interface{}{
+			"name":      name.Name,
+			"namespace": name.Namespace,
+		},
+		"spec": spec,
+	}
+	return CreateUnstructured(raw)
+}
+
+func GetManilaAPI(name types.NamespacedName) *manilav1.ManilaAPI {
+	instance := &manilav1.ManilaAPI{}
+	Eventually(func(g Gomega) {
+		g.Expect(k8sClient.Get(ctx, name, instance)).Should(Succeed())
+	}, timeout, interval).Should(Succeed())
+	return instance
+}
+
+func GetManilaScheduler(name types.NamespacedName) *manilav1.ManilaScheduler {
+	instance := &manilav1.ManilaScheduler{}
+	Eventually(func(g Gomega) {
+		g.Expect(k8sClient.Get(ctx, name, instance)).Should(Succeed())
+	}, timeout, interval).Should(Succeed())
+	return instance
+}
+
+func GetManilaShare(name types.NamespacedName) *manilav1.ManilaShare {
+	instance := &manilav1.ManilaShare{}
+	Eventually(func(g Gomega) {
+		g.Expect(k8sClient.Get(ctx, name, instance)).Should(Succeed())
+	}, timeout, interval).Should(Succeed())
+	return instance
+}
+
+func ManilaAPIConditionGetter(name types.NamespacedName) condition.Conditions {
+	instance := GetManilaAPI(name)
+	return instance.Status.Conditions
+}
+
+func ManilaAPINotExists(name types.NamespacedName) {
+	Consistently(func(g Gomega) {
+		instance := &manilav1.ManilaAPI{}
+		err := k8sClient.Get(ctx, name, instance)
+		g.Expect(k8s_errors.IsNotFound(err)).To(BeTrue())
+	}, timeout, interval).Should(Succeed())
+}
+
+func ManilaAPIExists(name types.NamespacedName) {
+	Consistently(func(g Gomega) {
+		instance := &manilav1.ManilaAPI{}
+		err := k8sClient.Get(ctx, name, instance)
+		g.Expect(k8s_errors.IsNotFound(err)).To(BeFalse())
+	}, timeout, interval).Should(Succeed())
+}
+
+func ManilaSchedulerExists(name types.NamespacedName) {
+	Consistently(func(g Gomega) {
+		instance := &manilav1.ManilaScheduler{}
+		err := k8sClient.Get(ctx, name, instance)
+		g.Expect(k8s_errors.IsNotFound(err)).To(BeFalse())
+	}, timeout, interval).Should(Succeed())
+}
+
+func ManilaShareExists(name types.NamespacedName) {
+	Consistently(func(g Gomega) {
+		instance := &manilav1.ManilaShare{}
+		err := k8sClient.Get(ctx, name, instance)
+		g.Expect(k8s_errors.IsNotFound(err)).To(BeFalse())
+	}, timeout, interval).Should(Succeed())
+}
+
+func ManilaSchedulerNotExists(name types.NamespacedName) {
+	Consistently(func(g Gomega) {
+		instance := &manilav1.ManilaScheduler{}
+		err := k8sClient.Get(ctx, name, instance)
+		g.Expect(k8s_errors.IsNotFound(err)).To(BeTrue())
+	}, timeout, interval).Should(Succeed())
+}
+
+func ManilaShareNotExists(name types.NamespacedName) {
+	Consistently(func(g Gomega) {
+		instance := &manilav1.ManilaShare{}
+		err := k8sClient.Get(ctx, name, instance)
+		g.Expect(k8s_errors.IsNotFound(err)).To(BeTrue())
+	}, timeout, interval).Should(Succeed())
+}
+
+func SimulateDeploymentReadyWithPods(name types.NamespacedName, networkIPs map[string][]string) {
```

Comment:
I think this is a good time to move this to lib-common

Reply:
Agree, and trying to address it here: https://github.com/openstack-k8s-operators/lib-common/pull/274

Reply:
I'm going to update this patch as long as the lib-common change lands

Reply:
Thank you for pulling this out. I merged the lib-common patch.